utilities/unittests/test_pytest_utils.py

```
@@ -2493,6 +2494,81 @@ def test_empty_marker_expression_filters_hpp(self):
+        assert result == [item_other]
+        config.hook.pytest_deselected.assert_called_once_with(items=[item_post_test_alerts])
+
+    def test_no_filtering_when_no_flags_set(self):
+        """All items are returned unchanged when no filtering flags are set."""
+        item_post_test_alerts = MagicMock()
+        item_post_test_alerts.keywords = {"post_test_alerts": True}
+        item_other = MagicMock()
+        item_other.keywords = {"other_test": True}
+        items = [item_post_test_alerts, item_other]
+        config = MagicMock()
+        config.getoption.side_effect = lambda flag: False
```

Ruff (0.15.21)

[warning] Unused lambda argument: `flag`

(ARG005)